### PR TITLE
[Merged by Bors] - layerfetcher: syntactically invalid blocks are expected from malicious peers

### LINKED ISSUE #2697: syncer fails sync too aggressively on invalid blocks
## Description

when debugging late-nodes system test failure on develop branch, one of the issues that came up was that syncer fails sync as soon as block fails syntactic validity check (caused by nodes disagreeing on different tortoise beacon value).

this behavior is too aggressive, as syntactically invalid blocks are expected from malicious peers.

## Expected Behavior

when syncer 
- gets database.ErrNotFound from peer, it should retry to get blocks from other peers until it exhausted all peers
- receives an syntactically invalid block, it should skip it instead of failing sync

### PR DESCRIPTION
**Pull request by evt**
## Motivation
Closes #2697 

## Changes
Skip returning error for invalid blocks

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Review by dshulyak — APPROVED** [review]

**Comment by evt** [comment]
bors try

File: layerfetcher/layers.go [hunk 1]
```diff
@@ -338,8 +338,9 @@ func (l *Logic) fetchLayerData(ctx context.Context, logger log.Log, layerID type
 	}
 	logger.With().Debug("fetching new blocks", log.Int("to_fetch", len(blocksToFetch)))
 	if err := l.GetBlocks(ctx, blocksToFetch); err != nil {
-		// fail sync for the entire layer
-		return err
+		logger.With().Error("failed fetching new blocks", log.Err(err))
+		// syntactically invalid blocks are expected from malicious peers
+		return nil
```
Comment (countvonzero) [inline layerfetcher/layers.go]: no need to return. just fall through here.

File: layerfetcher/layers.go [hunk 2]
```diff
@@ -338,8 +338,9 @@ func (l *Logic) fetchLayerData(ctx context.Context, logger log.Log, layerID type
 	}
 	logger.With().Debug("fetching new blocks", log.Int("to_fetch", len(blocksToFetch)))
 	if err := l.GetBlocks(ctx, blocksToFetch); err != nil {
-		// fail sync for the entire layer
-		return err
+		logger.With().Error("failed fetching new blocks", log.Err(err))
```
Comment (countvonzero) [inline layerfetcher/layers.go]: Warn instead of Error.

File: layerfetcher/layers_test.go [hunk 3]
```diff
@@ -352,9 +352,14 @@ func TestPollLayerBlocks_FetchLayerBlocksError(t *testing.T) {
...
 			return map[types.Hash32]chan fetch.HashDataPromiseResult{types.RandomHash(): ch}
 		}).Times(numPeers)
+	tl.mLayerDB.EXPECT().SaveHareConsensusOutput(gomock.Any(), layerID, gomock.Any()).DoAndReturn(
+		func(_ context.Context, _ types.LayerID, blockID types.BlockID) interface{} {
+			assert.NotEqual(t, blockID, types.EmptyBlockID)
+			return nil
+		}).Times(1)
 
 	res := <-tl.PollLayerContent(context.TODO(), layerID)
-	assert.Equal(t, ErrLayerDataNotFetched, res.Err)
```
Comment (countvonzero) [inline layerfetcher/layers_test.go:357]: can you change the test name to `FetchLayerBlocksErrorIgnored`? thanks

**Review by countvonzero — APPROVED** [review]

File: layerfetcher/layers_test.go (hunk 3 above)
Comment (evt) [reply layerfetcher/layers_test.go:357]: done

File: layerfetcher/layers.go (hunk 2 above)
Comment (evt) [reply layerfetcher/layers.go]: done

File: layerfetcher/layers.go (hunk 1 above)
Comment (evt) [reply layerfetcher/layers.go]: done

**Comment by evt** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Pull request closed without merging**